pkg/scalers/datadog_scaler.go

```diff
@@ -0,0 +1,316 @@
+
+	if len(series) == 0 {
+		return false, nil
+	}
+
+	points := series[0].GetPointlist()
+
+	if len(points) == 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// getQueryResult returns result of the scaler query
+func (s *datadogScaler) getQueryResult(ctx context.Context) (int, error) {
+
+	ctx = context.WithValue(
+		context.Background(),
+		datadog.ContextAPIKeys,
+		map[string]datadog.APIKey{
+			"apiKeyAuth": {
+				Key: s.metadata.apiKey,
+			},
+			"appKeyAuth": {
+				Key: s.metadata.appKey,
+			},
+		},
+	)
+
+	ctx = context.WithValue(ctx,
+		datadog.ContextServerVariables,
+		map[string]string{
+			"site": s.metadata.datadogSite,
+		})
+
+	resp, _, err := s.apiClient.MetricsApi.QueryMetrics(ctx, time.Now().Unix()-int64(s.metadata.age), time.Now().Unix(), s.metadata.query)
+	if err != nil {
+		return -1, fmt.Errorf("error when retrieving Datadog metrics: %s", err)
+	}
+
+	series := resp.GetSeries()
+
+	if len(series) == 0 {
+		return 0, fmt.Errorf("no Datadog metrics returned")
+	}
+
+	points := series[0].GetPointlist()
+
+	if len(points) == 0 {
+		return 0, fmt.Errorf("no Datadog metrics returned")
+	}
+
+	var val int = int(*points[0][1])
+
+	return val, nil
+}
+
+// GetMetricSpecForScaling returns the MetricSpec for the Horizontal Pod Autoscaler
+func (s *datadogScaler) GetMetricSpecForScaling(context.Context) []v2beta2.MetricSpec {
+	var externalMetric *v2beta2.ExternalMetricSource
+
+	targetQueryValue := resource.NewQuantity(int64(s.metadata.queryValue), resource.DecimalSI)
+
+	switch s.metadata.vType {
+
+	case average:
+		externalMetric = &v2beta2.ExternalMetricSource{
+			Metric: v2beta2.MetricIdentifier{
+				Name: s.metadata.metricName,
+			},
+			Target: v2beta2.MetricTarget{
+				Type:         v2beta2.AverageValueMetricType,
+				AverageValue: targetQueryValue,
+			},
+		}
+	case global:
+		externalMetric = &v2beta2.ExternalMetricSource{
+			Metric: v2beta2.MetricIdentifier{
+				Name: s.metadata.metricName,
+			},
+			Target: v2beta2.MetricTarget{
+				Type:  v2beta2.ValueMetricType,
+				Value: targetQueryValue,
+			},
+		}
+	}
+	metricSpec := v2beta2.MetricSpec{
+		External: externalMetric, Type: externalMetricType,
+	}
+	return []v2beta2.MetricSpec{metricSpec}
+}
+
+// GetMetrics returns value for a supported metric and an error if there is a problem getting the metric
+func (s *datadogScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	num, err := s.getQueryResult(ctx)
+	if err != nil {
+		datadogLog.Error(err, "error getting metrics from Datadog")
+		return []external_metrics.ExternalMetricValue{}, fmt.Errorf("error getting metrics from Datadog: %s", err)
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: s.metadata.metricName,
+		Value:      *resource.NewQuantity(int64(num), resource.DecimalSI),
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
```

Comment:
```suggestion
	return []external_metrics.ExternalMetricValue{metric}, nil
```

Reply:
mmm, I don't think that change is correct. GetMetrics needs to return a slice of external_metrics.ExternalMetricValue struct, which is what I am doing there. Am I missing anything?

Reply:
@arschles ? ☝️

Reply:
I think it is ok to keep it this way :)

Reply:
sorry for the delay! I lost track of this PR.

the `append` is technically correct. I was just suggesting the change because it avoids the extra `append`.